django/db/models/sql/compiler.py

```diff
@@ -1647,9 +1647,7 @@ def prepare_value(self, field, value):
                     "Window expressions are not allowed in this query (%s=%r)."
                     % (field.name, value)
                 )
-        else:
-            value = field.get_db_prep_save(value, connection=self.connection)
-        return value
+        return field.get_db_prep_save(value, connection=self.connection)
```

Comment:
These changes make it so compilable (`hasattr(value, "as_sql")`) can now make their way to `get_db_prep_save` but it shouldn't be a complete novelty as _compilable_ could already make their way to `get_db_prep` (e.g. see `FieldGetDbPrepValueMixin.get_db_prep_lookup`.